### ISSUE TITLE
Does this YAML work with the latest LilyGo T5 and touchscreen?

### ISSUE THREAD

**Issue by ZogKarndon** [Januznl/esphome-lilygo-t5-s3-plus#1]
I've got a LilyGo "Screen 4.7-S3 V 2.3" with an SD card interface that I'm trying to get working with ESPHome 2023.12.5.

When I flash the image via ESPHome Web, the device goes into a boot loop:
```
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x8 (TG1WDT_SYS_RST),boot:0x2c (SPI_FAST_FLASH_BOOT)
Saved PC:0x40377e36
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x43c
load:0x403c9700,len:0xbec
load:0x403cc700,len:0x2a3c
entry 0x403c98d8
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x8 (TG1WDT_SYS_RST),boot:0x2c (SPI_FAST_FLASH_BOOT)
Saved PC:0x40377e36
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x43c
load:0x403c9700,len:0xbec
load:0x403cc700,len:0x2a3c
entry 0x403c98d8
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x8 (TG1WDT_SYS_RST),boot:0x2c (SPI_FAST_FLASH_BOOT)
Saved PC:0x40377e36
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x43c
load:0x403c9700,len:0xbec
load:0x403cc700,len:0x2a3c
entry 0x403c98d8
E (5893) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
E (5893) task_wdt:  - IDLE (CPU 0)
E (5893) task_wdt: Tasks currently running:
E (5893) task_wdt: CPU 0: loopTask
E (5893) task_wdt: CPU 1: IDLE
E (5893) task_wdt: Aborting.
```

**Comment by Januznl** [comment]
Did you connected the battery while this issue occurred? 
Have you tried leaving it powerless for 10/15min and try powering it on through USB with the ESPHome Web log?

Mine had this issue after i disconnected the battery and reconnecting it, leaving it for a while fixed it.

**Comment by ZogKarndon** [comment]
I don't currently have a battery attached; it's running entirely via USB.  I get the logs via the ESPHome web log.

**Comment by Januznl** [comment]
Hmm ok, it's such a weird issue.
Mainly because it's fixed on my board, but I'm still not sure why.


Have you tried another usb cable?

**Comment by ZogKarndon** [comment]
I did try another USB cable, with no difference.   Just to double check - it's [this ](https://www.lilygo.cc/cdn/shop/products/T5-S3V4.7.jpg?v=1665121576&width=990)[board](https://www.lilygo.cc/products/t5-4-7-inch-e-paper-v2-3), right?

**Comment by Januznl** [comment]
Yes, it is specific for this board, the V2.3 / Plus with the ESP32-S3 processor.
This repo is based on the following thread, someone else also followed this repo and got it to work.  (see https://github.com/esphome/feature-requests/issues/1960#issuecomment-1871354283 )

btw, are you able to install the demo with the arduino IDE?
That is also something i did in the meanwhile until it worked.
Maybe this demo will install something into the bootloader on the board?

arduino demo: https://github.com/Xinyuan-LilyGO/LilyGo-EPD47?tab=readme-ov-file

**Comment by ZogKarndon** [comment]
I installed the [minimal demo](https://github.com/Xinyuan-LilyGO/LilyGo-EPD47/tree/master/examples/demo) using the Arduino 2.2.1 IDE, and it uploaded & ran fine.   (I double checked the configuration to make sure it matched the [Lilygo docs](https://t5-47-t5-47-plus.readthedocs.io/en/latest/get-started/index.html#compile-on-arduino)

*BUT* when I tried the [touchtest demo](https://github.com/Xinyuan-LilyGO/LilyGo-EPD47/tree/master/examples/touchtest), it compiled but went into a boot loop similar to what I'm seeing with your sample:
```
14:22:24.746 -> ESP-ROM:esp32s3-20210327
14:22:24.746 -> Build:Mar 27 2021
14:22:24.746 -> rst:0xc (RTC_SW_CPU_RST),boot:0xc (SPI_FAST_FLASH_BOOT)
14:22:24.746 -> Saved PC:0x40377f18
14:22:24.746 -> SPIWP:0xee
14:22:24.746 -> mode:DIO, clock div:1
14:22:24.746 -> load:0x3fce3808,len:0x43c
14:22:24.746 -> load:0x403c9700,len:0xbec
14:22:24.746 -> load:0x403cc700,len:0x2a3c
14:22:24.785 -> SHA-256 comparison failed:
14:22:24.785 -> Calculated: 74cb8a0835ff948a23b1fa30f5641bd3d3fc50d3e848150763aab2d791fb9d34
14:22:24.785 -> Expected: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
14:22:24.785 -> Attempting to boot anyway...
14:22:24.785 -> entry 0x403c98d8
14:22:29.899 -> E (10146) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
14:22:29.899 -> E (10146) task_wdt:  - IDLE (CPU 0)
14:22:29.899 -> E (10146) task_wdt: Tasks currently running:
14:22:29.899 -> E (10146) task_wdt: CPU 0: loopTask
14:22:29.899 -> E (10146) task_wdt: CPU 1: IDLE
14:22:29.899 -> E (10146) task_wdt: Aborting.
```

Edit:  Also tried the [wifi_sync demo](https://github.com/Xinyuan-LilyGO/LilyGo-EPD47/tree/master/examples/wifi_sync), and *that* worked.... Hm...

**Comment by Januznl** [comment]
I don't have a touchscreen, could you test this demo without the touchscreen attached? maybe one of the sub-packages is causing an issue with the touchscreen.

**Comment by ZogKarndon** [comment]
Tried a couple of other demos, along with the OpenWeatherMap sample, and could get those to work.  (Never did get touch-test to work, though.)

**Comment by Januznl** [comment]
Did you managed to get this repo working without the touchscreen?

**Comment by AStoker** [comment]
Running into this exact issue. I'm not using the touchscreen. Oddly enough, if I flash it with esphome locally, it seems to work fine. But when I try to flash it with esphome on home assistant, it fails. I've yet to figure out the difference in my file configurations yet...
